### ISSUE TITLE
Azure - Non existing metric mapped to a random one.

### ISSUE THREAD

**Issue by ivan-shaporov** [capitalone/cloud-custodian#2792]
Tried running this policy:
 ```
policies:
    - name: bad-metric
      description: |
        Non existing metric mapped to a random one.
      resource: azure.vm
      
      filters:
      - type: metric
        metric: NonExistingMetric
        aggregation: average
        op: ne
        threshold: 17
        timeframe: 12

      actions:
       - type: tag
         tag: bug
         value: NonExistingMetric
 ```
Expected an error message, got the tag assigned.

Also tried le 0.75 condition, got one instance assigned the tag and one not. Looks like if a metric is not found a random metric is picked up.

VM OS - Linux, Cent OS.

**Comment by ivan-shaporov** [comment]
Created a ticket for the Azure REST API:

I am running the script provided below according to 
https://docs.microsoft.com/en-us/rest/api/monitor/metrics/list

Expecting to get Network metrics, but getting CPU metrics.

script:

AAD_TOKEN=`az account get-access-token --resource "https://management.azure.com/" --query accessToken -o tsv`

curl -H "Authorization: Bearer $AAD_TOKEN" https://management.azure.com/subscriptions/d36d0808-a967-4f73-9fdc-32ea232fc81d/resourceGroups/IVANCUSTODIAN/providers/Microsoft.Compute/virtualMachines/ivancustodian1/providers/microsoft.insights/metrics?api-version=2018-01-01&metricnames=Network+In


response:

{"cost":0,"timespan":"2018-08-23T18:44:48Z/2018-08-23T19:44:48Z","interval":"PT1M","value":[{"id":"/subscriptions/d36d0808-a967-4f73-9fdc-32ea232fc81d/resourceGroups/IVANCUSTODIAN/providers/Microsoft.Compute/virtualMachines/ivancustodian1/providers/Microsoft.Insights/metrics/Percentage CPU","type":"Microsoft.Insights/metrics","name":{"value":"Percentage CPU","localizedValue":"Percentage CPU"}, ... truncated

**Comment by ivan-shaporov** [comment]
This might be python SDK issue, repro:

from azure.common.client_factory import get_client_from_cli_profile
from azure.mgmt.monitor import MonitorManagementClient

monitorClient = get_client_from_cli_profile(MonitorManagementClient)

subscription = "{subscription}"
resourceGroup = "{resourceGroup]"
vmname = "{vm name}"

resource = "subscriptions/{}/resourceGroups/{}/providers/Microsoft.Compute/virtualMachines/{}" . format(subscription, resourceGroup, vmname)

metrics_data = monitorClient.metrics.list(resource, metric="NonExistingMetric", raw=True)

metricId = metrics_data.output.value[0].id

print(metricId)

expected error message, actual:
/subscriptions/{sub}/resourceGroups/{rg}/providers/Microsoft.Compute/virtualMachines/{vm}/providers/Microsoft.Insights/metrics/Percentage CPU

**Comment by aluong** [comment]
PR #2808

**Comment by ivan-shaporov** [comment]
Fixed with PR #2808
